From 0db4066637df67cd5ec26491a5b42a996ceb51cd Mon Sep 17 00:00:00 2001
From: changbetty <changbetty@google.com>
Date: Wed, 21 Dec 2022 10:16:10 +0000
Subject: [PATCH] Change the intent flag to immutable

Bug: 256590210
Test: make RunSettingsRoboTests ROBOTEST_FILTER=MediaVolumePreferenceControllerTest
Change-Id: I68132f863027fab19351b4fdbc363e5c274e3327
Merged-In: I68132f863027fab19351b4fdbc363e5c274e3327
(cherry picked from commit 0dcd6a5492fae67e0d7f1fb83f6d4db7eaa15b55)
Merged-In: I68132f863027fab19351b4fdbc363e5c274e3327
---
 .../notification/MediaVolumePreferenceController.java         | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/notification/MediaVolumePreferenceController.java b/src/com/android/settings/notification/MediaVolumePreferenceController.java
index 2466b7747a..85c053e3b0 100644
--- a/src/com/android/settings/notification/MediaVolumePreferenceController.java
+++ b/src/com/android/settings/notification/MediaVolumePreferenceController.java
@@ -124,7 +124,7 @@ public class MediaVolumePreferenceController extends VolumeSeekBarPreferenceCont
                     getWorker().getActiveLocalMediaController().getPackageName());
 
             pi = PendingIntent.getBroadcast(context, 0 /* requestCode */, intent,
-                    PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_MUTABLE);
+                    PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
         } else {
             final CachedBluetoothDevice bluetoothDevice =
                     ((BluetoothMediaDevice) mMediaDevice).getCachedDevice();
@@ -141,7 +141,7 @@ public class MediaVolumePreferenceController extends VolumeSeekBarPreferenceCont
                     && getWorker().getActiveLocalMediaController() != null);
 
             pi = PendingIntent.getActivity(context, 0 /* requestCode */, intent,
-                    PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_MUTABLE);
+                    PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
         }
 
         final IconCompat icon = getBroadcastIcon(context);
-- 
2.17.1

